internal/controller/checkpointrestoreoperator_controller.go

```diff
@@ -45,11 +45,30 @@ import (
 	"github.com/fsnotify/fsnotify"
 )
 
+type DeletionPolicy int
+
+const (
+	ByCount DeletionPolicy = iota
+	BySize
```

Comment:
It might be useful to define two types of retention policies (_count-based_ and _age-based_) and allow users to specify a _storage quota_ separately.

- For policies with count-based retention, users need to specify a threshold, and the operator will automatically delete the oldest checkpoint when the specified threshold is met.
- For age-based policies, users specify a time period for which to retain checkpoints, and checkpoints are automatically deleted when they reach that age.

Reply:
@rst0git yes we already planed the age-based policies as Time-Based Checkpoint Retention Policy and I also mentioned it in my proposal, we will definitely implement that, but here I am using `ByCount` and `BySize` to decide when to stop deleting the checkpoints that are collected for count-based and storage-based policies respectively as both policies have different type of threshold.

Reply:
@rst0git yes I will update those, as the purpose of this draft PR was to make sure that I'm on right track, I didn't add those. Now as it's already clear I will update those with right description and messages. 

also I want to get some advice from you about the approach of detecting kubernetes checkpoint creation from our operator(we had a short discussion on this on gsoc element channel). What do you think about the current `fsnotify` solution, @adrianreber was not sure about if this is the best solution. 

- Do we have any other better solutions than this? 

- Where does kubernetes store logs when a checkpoint is created? can we parse those logs to check checkpoint creation??

Reply:
If checkpointing will be an API server operation at some point, maybe, we might get notifications. Parsing log files does not sound like a reliable way.